IntegrationTests/UITests/UITest.cs

```diff
@@ -38,23 +39,61 @@ void HandleApplicationIdle(object sender, EventArgs e)
             {
                 await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                 await WaitForIdleAsync();
-                var dialog = Application.OpenForms.OfType<T>().Single();
+                using var form = Application.OpenForms.OfType<T>().Single();
+                Assert.True(form.Visible, $"{form.GetType().FullName} {form.Name} should be visible");
                 try
                 {
-                    await runAsync(dialog);
+                    await runTestAsync(form);
                 }
                 finally
                 {
-                    dialog.Close();
+                    form.Close();
                 }
             });
 
-            showDialog();
+            showForm();
 
             // Join the asynchronous test operation so any exceptions are rethrown on this thread
             test.Join();
         }
 
+        public static void RunDialog<T>(
+            Action<Form> showDialog,
+            Func<T, Task> runTestAsync)
+            where T : Form
+        {
+            using var mainForm = new Form { Text = $"Test {typeof(T).Name}" };
+            RunForm<T>(
+                showForm: () =>
+                {
+                    mainForm.Shown += (s, e) =>
+                    {
+                        showDialog(mainForm);
+                    };
+
+                    if (Application.MessageLoop)
+                    {
+                        mainForm.Show(owner: null);
+                    }
+                    else
+                    {
+                        Application.Run(mainForm);
+                    }
+                },
+                runTestAsync: async (dialogForm) =>
+                {
+                    try
+                    {
+                        await runTestAsync(dialogForm);
+                    }
+                    finally
+                    {
+                        // dispose the mainForm in the main thread
+                        mainForm.Dispose();
```

Comment:
The whole test runs on the main thread, so this is just duplicating the call that will occur for the `using` block.

Reply:
📝 The `runTestAsync` argument can be passed without changes:

```
runTestAsync: runTestAsync
```